test/sys/test_signalfd.rs

```diff
@@ -0,0 +1,52 @@
+#[cfg(target_os = "linux")]
+#[test]
+// FIXME: Fails with "fatal runtime error: failed to initiate panic, error 5" on i686-unknown-linux-musl
+#[cfg_attr(all(target_env = "musl", target_arch = "x86"), ignore)]
+fn test_signalfd() {
+    use nix::sys::signalfd::{SFD_NONBLOCK, SignalFd};
+    use nix::sys::signal;
+    use nix::unistd;
+    use nix::unistd::*;
+    use nix::unistd::ForkResult::*;
+    use nix::sys::signal::*;
+    use nix::sys::wait::*;
+
+    // Set up a signalfd to listen for SIGUSR1 on this thread
+    let mut mask = SigSet::empty();
+    mask.add(SIGUSR1);
+    mask.thread_block().unwrap();
+
+    // Start a child process
+    match unistd::fork() {
```

Comment:
`ECHILD` is an errno, not a signal.  Signal 10, on Linux, is `SIGUSR1`.  It sounds like you weren't correctly blocking the signal.  I can take a look at it, if you want.